src/goto-harness/goto_harness_parse_options.cpp

```diff
@@ -87,13 +124,15 @@ int goto_harness_parse_optionst::doit()
   harness_generator->generate(
     goto_model, got_harness_config.harness_function_name);
 
-  // Write end result to new goto-binary
-  if(write_goto_binary(
-       got_harness_config.out_file, goto_model, ui_message_handler))
-  {
-    throw system_exceptiont{"failed to write goto program from file '" +
-                            got_harness_config.out_file + "'"};
-  }
+  filter_goto_model(goto_model, goto_model_without_harness_symbols);
+  auto harness_out = std::ofstream{got_harness_config.out_file};
+  dump_c(
+    goto_model.goto_functions,
+    true,
+    true,
+    false,
```

Comment:
:pick: Pull these out into named local variables so there is some clue as to what they match up to. E.g. 

```C++
const bool use_system_headers = true;
...
dump_c(
  goto_model.goto_functions, 
  use_system_headers, 
...
```